dart/lib/src/browser_client.dart

```diff
@@ -41,19 +42,19 @@ class SentryBrowserClient extends SentryClient {
     );
   }
 
-  SentryBrowserClient._(SentryOptions options, {String origin, String platform})
+  SentryBrowserClient._(SentryOptions options,
+      {String origin, @required String platform})
       : super.base(
           options,
-          origin: origin,
-          sdk: Sdk(name: browserSdkName, version: sdkVersion),
-          platform: platform,
+          transport: Transport(
+            dsn: options.dsn,
+            httpClient: options.httpClient,
+            clock: options.clock,
+            compressPayload: false,
+            sdk: Sdk(name: browserSdkName, version: sdkVersion),
+            headersBuilder: SentryClient.buildHeaders,
+            platform: platform,
```

Comment:
wondering if this is the same as `options.sentryClientName`, could you check it?

Reply:
`platform` is 'dart' or 'javascript' , according to the docs `options.sentryClientName` is 'sentry.{language}.{platform}/{version}'

Reply:
we could do this then?

```
factory SentryIOClient(SentryOptions options) {
    options.platform = sdkPlatform;
    SentryIOClient._(options);
  }
```

same for the WebClient, so we move this flag to the options and the `transport` don't need to take it as a param? we do it automatically.

Reply:
✅  platform is now defined at runtime